pkgs/development/libraries/aqbanking/default.nix

```diff
@@ -1,31 +1,45 @@
-{ stdenv, fetchurl, gmp, gwenhywfar, libtool, libxml2, libxslt
-, pkgconfig, gettext, xmlsec, zlib
+{ stdenv, fetchFromGitHub, gmp, gwenhywfar, libtool, libxml2, libxslt
+, pkgconfig, gettext, xmlsec, zlib, autoreconfHook
 }:
 
-let
-  inherit ((import ./sources.nix).aqbanking) sha256 releaseId version;
-in stdenv.mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "aqbanking";
-  inherit version;
-
-  src = let
-    qstring = "package=03&release=${releaseId}&file=02";
-    mkURLs = map (base: "${base}/sites/download/download.php?${qstring}");
-  in fetchurl {
-    name = "${pname}-${version}.tar.gz";
-    urls = mkURLs [ "http://www.aquamaniac.de" "http://www2.aquamaniac.de" ];
-    inherit sha256;
+  version = "5.99.40beta";
+
+  src = fetchFromGitHub {
+    owner = "aqbanking";
+    repo = "aqbanking";
+    rev = version;
+    sha256 = "1p80cmzy2c0k17vknpsywr9ywimak09y3lvyhvc3dyqfcr0xnbm9";
```

Comment:
Why not use something like https://www.aquamaniac.de/rdm/attachments/download/207/aqbanking-5.99.41beta.tar.gz (btw. 5.99.41 is out) here? The tarballs seem to have stable links and even if they become unstable, we have a tarball mirror at https://tarballs.nixos.org/, using a mirror from someone other than the original maintainer really opens up another possibility where malicious code can be injected (especially if you're doing your online banking).

There is also another issue with beta versions, because as far as I can see only the stable releases are signed and I usually verify these signatures (in this case manuall) when updating, so what's the reason why kmymoney wants the beta?

Reply:
I would really like to have this PR go through, too :wink: 

The link per se is stable but their server is down pretty often (multiple times per hour, each for several minutes—at least when I tried to create this PR myself two week ago).

We need the beta because of PSD2 being in effect; IIRC the non-beta builds do not yet support the new interface (which makes them entirely unusable, as in, you can't connect to any bank).

Reply:
@aszlig 
When creating the pr upstream wasn't reachable at all.
Now it's better and i updated the pr to reflect your concerns about 3rd party sources.
I also have the impression that PSD2 is only supported by the beta versions.
I understand your concerns about using beta releases.
But not being able to use aqbanking at all is worse than not being able to verify the signature in my opinion.
When there is a usable non-beta release we should of course go back to using that.